Updating EIP 2515 for clarity. (#2605)
ethereum/EIPs: EIPS/eip-2515.md

---
eip: 2515
title: Update Difficulty Bomb with a Freezing Mechanism
author: James Hancock (@madeoftin)
discussions-to: https://ethereum-magicians.org/t/eip-2515-replace-the-difficulty-bomb-with-a-difficulty-freeze/3995
status: Draft
type: Standards Track
category: Core

> created: 2020/02/10

Reviewer: ```suggestion
created: 2020-02-10
```

The Difficulty Freeze is an update to the Difficulty Bomb implemented within the Ethereum protocol's difficulty adjustment algorithm. The difficulty would freeze at a specific block height, determined in advance, and increases by 0.01% per block following that block height. This does not stop the chain, but it keeps the functionality of the original difficulty bomb, but makes it simple to predict when the effect occurs. 

## Abstract
<!--A short (~200 word) description of the technical issue being addressed.-->
The difficulty freeze mechanism is easy to predict and model. The client maintainers are also positioned to respond to an incoming Difficulty Freeze because they are the arbiters of client release code. This, combined with the predictability, will lead to more timely diffusion of the bomb.

## Motivation
<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->

> The current difficulty bombs' effect on the Block Time Targeting mechanism is rather complex to model, and it has both appeared when it was not expected (Muir Glacier) and negatively affected users and miners when they are not the target (in the case of delaying forks due to technical difficulties). Users are affected as a function of the usability of the chain is affected by increased block times. Miners are affected by a reduction in block rewards due to the increase in block time. Both of these groups are unable on their own to address the difficulty bomb. In the case of the Difficulty Freeze it is more predictiable and so knowing when to make the change is readily apparent at all times.

Reviewer: There is a single difficulty bomb:

```suggestion
The current difficulty bomb's effect on the Block Time Targeting mechanism is rather complex to model, and it has both appeared when it was not expected (Muir Glacier) and negatively affected users and miners when they are not the target (in the case of delaying forks due to technical difficulties). Users are affected as a function of the usability of the chain is affected by increased block times. Miners are affected by a reduction in block rewards due to the increase in block time. Both of these groups are unable on their own to address the difficulty bomb. In the case of the Difficulty Freeze it is more predictable and so knowing when to make the change is readily apparent at all times.
```

---

> eip: 2512

Reviewer: How was this merged with the wrong number in the first place?

Author: No idea. It confuses me.

Block height is very easy to predict and evaluate within the system. This removes the effect of the Difficulty Bomb on block time, simplifying the block time targeting mechanism. The addition of an increase in the difficulty was added after feedback that the game theory of the mechanism did not reliably result in planned obsolescence of the chain. It was still theoretically possible that a chain could continue in a zombie state with miners self-regulating hashrate to keep the chain at a semi-functional state. While the game theory tended towards miners reaching blocktimes too fast for the network to propagate, the addition of a steady increase in difficulty removes this possibility and is therefore more secure.

https://twitter.com/quentinc137/status/1227110578235330562

> ### Rational behind the 0.001% Increase

Reviewer: ```suggestion
### Rationale behind the 0.001% Increase
```

Block height is very easy to predict and evaluate within the system. This removes the effect of the Difficulty Bomb on block time, simplifying the block time targeting mechanism. The addition of an increase in the difficulty was added after feedback that the game theory of the mechanism did not reliably result in planned obsolescence of the chain. It was still theoretically possible that a chain could continue in a zombie state with miners self-regulating hashrate to keep the chain at a semi-functional state. While the game theory tended towards miners reaching blocktimes too fast for the network to propagate, the addition of a steady increase in difficulty removes this possibility and is therefore more secure.

https://twitter.com/quentinc137/status/1227110578235330562

> __Rational behind 0.001% Increase__

Reviewer: Why not use markdown headings?

```suggestion
### Rational behind the 0.001% Increase
```

> The algorithm is not purely linear, but slowly increases exponentially every block by .01% of the previous. It still needs to be determined if a pure linear function is prefered, or which value of the scalar if it is exponential.

Reviewer: ```suggestion
The algorithm is not purely linear, but slowly increases exponentially every block by .01% of the previous. It still needs to be determined if a pure linear function is preferred, or which value of the scalar if it is exponential.
```

prefered -> preferred typo, which makes the build fail
